### PR TITLE
Sass json

### PR DESCRIPTION
**Pull request by hippogriffic**
the grunt tasks were taken from another project and were not written by me

everything works and compiles correctly

@incuna/frontend please review

### CODE REVIEW

File: Gruntfile.js [hunk 1]
```diff
@@ -97,17 +144,42 @@ module.exports = function (grunt) {
...
         'sass',
         'nunjucks',
         'watch'
     ]);
 
     grunt.registerTask('build', [
+        'icons',
+        'svgstore',
+        'json-to-sass',
+        'json-to-sass-map',
```
Comment (Minglee01) [inline Gruntfile.js]: Can these 4 be ran seperately. cause u could move into a seperate task and call that here and in the dev task.

File: package.json [hunk 2]
```diff
@@ -27,6 +27,7 @@
   },
   "dependencies": {
     "grunt-nunjucks-2-html": "3.0.0",
+    "grunt-svgstore": "^1.0.0",
```
Comment (Minglee01) [inline package.json]: Do we need a strict version numner?

File: scripts/grunt-config.js [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+            config: {
+                paths: {
+                    iconsDir: './icons',
+                    jsonDir: './json',
+                    sassDir: './sass',
+                    cssDir: './styles',
+                    generatedSassDir: './sass/generated',
+                    templatesDir: './templates',
+                    scriptsDir: './scripts'
+
```
Comment (Minglee01) [inline scripts/grunt-config.js]: you could remove 
blankline 9 and 19
so it looks neater

File: package.json (hunk 2 above)
Comment (hippogriffic) [reply package.json]: whoops missed that

File: scripts/json-to-sass-map.js [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+        };
+        this.files.forEach((file) => {
+            let sassString = `// Compiled from ${file.src}`;
+            let src;
+            try {
+                src = JSON.parse(grunt.file.read(file.src));
+                sassString += compileObjectToSassMap(src);
+                grunt.file.write(file.dest, sassString);
+            } catch (error) {
+                grunt.log.error(`json-to-sass-map: cannot load src file ${file.src}`);
```
Comment (Minglee01) [inline scripts/json-to-sass-map.js:35]: Have you tried hitting this catch statement?

**Comment by Minglee01** [comment]
Good to merge

**Comment by hippogriffic** [comment]
@Minglee01 changes made :)

**Comment by hippogriffic** [comment]
thanks ⭐️

**Pull request merged**